ArduPlane/quadplane.cpp

```diff
@@ -2004,14 +2007,8 @@ bool QuadPlane::in_vtol_mode(void) const
     if (!available()) {
         return false;
     }
-    if (plane.control_mode == &plane.mode_qrtl &&
-        (poscontrol.get_state() == QPOS_APPROACH ||
-         poscontrol.get_state() == QPOS_AIRBRAKE)) {
-        return false;
-    }
-    if (in_vtol_land_approach() &&
-        poscontrol.get_state() == QPOS_APPROACH) {
-        return false;
+    if (in_vtol_land_sequence()) {
+        return poscontrol.get_state() != QPOS_APPROACH;
```

Comment:
This is certainly a change in behavior. The `QPOS_NONE` case might catch us out. 

I don't see any use of `in_vtol_mode` in this PR, maybe just a drive by? I would be tempted to leave it as it is. 

I'm also not sure what happens in the case of `plane.auto_state.vtol_loiter` in the check on line 2025. Because this now always returns if `in_vtol_land_sequence()` we miss some of the checks lower down. 

It might be fine, but there might also be a odd edge case, this code is all rather tricky.

Reply:
I don't think we can be in land sequence and have QPOS_NONE
vtol_loiter shouldn't matter in the landing sequence